File: models/Catalogue.js
class Catalogue {
    constructor() {
        this.books = []; // array of Book objects
    }

    addBook(book) {
        this.books.push(book);
    }

    getBook(id) {
        return this.books.find(b => b.id === id);
    }

    getAllBooks() {
        return this.books;
    }
}

module.exports = Catalogue;

File: src/main/java/org/example/library/Catalogue.java
package org.example.library;

import java.util.ArrayList;
import java.util.List;

public class Catalogue {
    ArrayList<Book> catalogue;

    public Catalogue() {
        catalogue = new ArrayList<>();
    }

    public void addBook(Book book) {
        catalogue.add(book);
    }

    public Book getBook(int index) {
        return catalogue.get(index);
    }

    public int getCatalogueSize() {
        return catalogue.size();
    }

    public List<Book> getBooks() {
        return catalogue;
    }

}
